Add renameIfExists method to fs

rosen-vladimirov opened the pull request:
Add renameIfExists method to fs. Used to skip ENOENT errors when rename is called directly. Renames specified file to the specified name only in case it exists.
Returns True in case of successful rename. False in case the file does not exist.

justcodebuilduser commented:
:heart:

teobugslayer commented:
This is non-atomic operation and subject to all race conditions imaginable. Perhaps better looking code:

```
try {
  fs.rename();
  return true
catch(e) {
  if (e.code === ENOENT) {
    return false;
  }
  throw e;
}
```

rosen-vladimirov commented:
@teobugslayer , you are correct, I've fixed the code and the tests are passing (I've added them in appbuilder-cli for the moment, but I'll move all `fs` tests here in a separate PR).

teobugslayer commented:
:+1:

justcodebuilduser commented:
:heart:
